documentation/general/torn-sdk.md

```diff
@@ -0,0 +1,109 @@
+
+One downside to this approach is that it is possible to end up with two VBCSCompiler server processes. Consider that when a solution has a mix of .NET SDK style projects and legacy projects then in a torn state the .NET SDK projects will use the .NET SDK compiler will legacy projects will use the Visual Studio compiler. While not a desirable outcome, it is a correct one. Customers who wish to only have one VBCSCompiler should correct the torn state in their build tools.
+
+### Visual Studio using Visual Studio Analyzers
+
+#### .NET SDK in box analyzers
+
+Analyzers which ship in the .NET SDK box will change to having a copy checked into Visual Studio. When .NET SDK based projects are loaded at design time, the Visual Studio copy of the analyzer will be loaded. Roslyn already understands how to prefer Visual Studio copies of analyzers. That work will need to be extended a bit but that is pretty straight forward code.
+
+This approach enables us to take actions like NGEN or R2R analyzers inside of Visual Studio. This is a long standing request from the Visual Studio perf team but very hard to satisfy in our current
+
+This approach is already taken by [the Razor generator][code-razor-vs-load]. This work was done for other reliability reasons but turned out working well for this scenario. There is initial upfront work to get the in box copy loading but it has virtually zero maintanence cost.
+
+This also means these analyzers can vastly simplify their development model by alwasy targeting the latest version of Roslyn. There is no more need to multi-target beacuse the version of the compiler the analyzer will be used with is known at ship time for all core build scenarios.
+
+This means that our design time experience can differ from our command line experience when customers are in a torn state. Specifically it's possible, even likely in some cases, that the diagonstics produced by design time builds will differ from command line builds. That is a trade off that we are willing to make. Customers who wish to have a consistent experience between design time should not operate in a torn state.
+
+#### NuGet based analyzers
+
+Analyzers which ship via NuGet will continue to following the existing [support policies][matrix-of-paine]. This means that they will either need to target a sufficiently old version of Roslyn or implement multi-targeting support.
+
+## Work Breakdown
+
+- [ ]: Flow the Micosoft.Net.Compilers.Toolset.Framework package to the .NET SDK
+- [ ]: Create a new package Micosoft.NETSdk.Compiler.Roslyn in .NET SDK.
+  - [ ]: The contents of this package will include the contents of the Microsoft.Net.Compilers.Toolset.Framework package
+  - [ ]: The contents will include a README.md stating the package is **not** supported for direct user consumption.
```

Comment:
Should the package be unlisted after upload? That allows restores but not most downloads/direct use.

Reply:
Unlisting is a good idea. I'm also considering changing the package such that it's not usable via `<PackageReference>`. That would _significantly_ reduce the incentive to download it.